pkg/aws/metadata/metadata.go

```diff
@@ -11,53 +11,76 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 package metadata
 
 import (
+	"context"
 	"fmt"
 	"io"
-	"net/http"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 )
 
-func getMetadata(name string) (string, error) {
-	resp, err := http.Get("http://169.254.169.254/latest/meta-data/" + name)
+func newClient() (*imds.Client, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
```

Comment:
Thinking about the `context.TODO()` a bit more...I think we should probably address this in another PR. Would you mind filing an issue to tackle this? I think we would want to have some sort of reasonable timeout, as we don't want to block indefinitely trying to fetch metadata.

Reply:
Hi @christarazi , I've created this issue. I'm not sure I've put all the elements you thought about.

Reply:
Looks good! :pray: